web-app/src/test/java/com/google/sps/data/QuadTreeTest.java

```diff
@@ -0,0 +1,78 @@
+package com.google.sps.data;
+
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.junit.Test;
+import org.junit.Assert;
+import org.junit.Before;
+import org.mockito.Mockito;
+import java.io.IOException;
+import java.util.ArrayList;
+
+@RunWith(JUnit4.class)
+public class QuadTreeTest extends Mockito {
+
+  private PoliceReport report1 = new PoliceReport(30.0, 15.0, "test", 1234567); // NE
+  private PoliceReport report2 = new PoliceReport(-30.0, 15.0, "test", 1234567); // SE
+  private PoliceReport report3 = new PoliceReport(30.0, -15.0, "test", 1234567); // NW
+  private PoliceReport report4 = new PoliceReport(-30.0, -15.0, "test", 1234567); // SW
+  private PoliceReport report5 = new PoliceReport(80.0, 45.0, "test", 1234567); // NE
+
+  ArrayList<PoliceReport> reportList = new ArrayList<PoliceReport>();
+  QuadTree tree;
+
+  @Before
+  public void setUp() {
+    reportList.add(report1);
+    reportList.add(report2);
+    reportList.add(report3);
+    reportList.add(report4);
+    reportList.add(report5);
+
+    tree = new QuadTree();
+  }
+
+  @Test
+  public void reallocateReportsCorrectly() throws IOException {
```

Comment:
Is there a reason why you want to make this test seperately? Wouldn't it be better to just leave reallocateReportsCorrectly() as a private function, and make two trees where one tree requires reallocation and one doesn't??

Reply:
I wanted to go for more granular tests, since a generic insert test might not tell us what exactly is failing, just that there's something wrong. But I do also really like the idea of being able to keep functions that ought to be private private.